server/neptune/workflows/internal/terraform/job/runner.go

```diff
@@ -83,6 +93,9 @@ func (r *JobRunner) Plan(ctx workflow.Context, localRoot *terraform.LocalRoot, j
 }
 
 func (r *JobRunner) Apply(ctx workflow.Context, localRoot *terraform.LocalRoot, jobID string, planFile string) error {
+	ctx = workflow.WithRetryPolicy(ctx, temporal.RetryPolicy{
+		NonRetryableErrorTypes: []string{TerraformClientErrorType, TimeoutErrorType},
```

Comment:
I'm curious if the retry policy `NonRetryableErrorTypes: []string{TerraformClientErrorType}` set in the Plan() method persist when we get to `Apply()`? If it does, maybe we don't need to add  `TerraformClientErrorType` to the retry policy here?

Reply:
It doesn't we're creating a function scoped workflow context

Reply:
Gotcha!